### ISSUE TITLE
createAccessPoint - Missing required header for this request: x-amz-content-sha256

### ISSUE THREAD

**Issue by ganeshnj** [awslabs/aws-sdk-swift#659]
### Describe the bug

```
2022-10-30T17:10:26-0700 info SerialExecutor : [Logging] Connection to endpoint: Optional("https://727033484140.s3-control.us-west-2.amazonaws.com/v20180820/accesspoint/jangirg-testing-ap?") is closing
Swift/ErrorType.swift:200: Fatal error: Error raised at top level: ClientRuntime.SdkError<AWSS3Control.CreateAccessPointOutputError>.client(ClientRuntime.ClientError.retryError(ClientRuntime.SdkError<AWSS3Control.CreateAccessPointOutputError>.service(AWSS3Control.CreateAccessPointOutputError.unknown(AWSClientRuntime.UnknownAWSHttpServiceError(_isThrottling: false, _statusCode: Optional(http_status_400), _headers: Optional(Server: AmazonS3, 
Content-Type: application/xml, 
x-amz-request-id: 5HMBDD2G2T1M7SDW, 
x-amz-id-2: XiUPE6xSNIdSgBWeh/JreYxEjXZlTkMEDoY24l8XyIC1b8eHZiuF0Y/vuSxhOvYUqCslVh8g3II=, 
Transfer-Encoding: chunked, 
Date: Mon, 31 Oct 2022 00:10:25 GMT, 
Connection: close), _message: Optional("Missing required header for this request: x-amz-content-sha256"), _requestID: Optional("5HMBDD2G2T1M7SDW"), _retryable: false, _type: ClientRuntime.ErrorType.unknown)), 
Status Code: http_status_400 
 x-amz-request-id: 5HMBDD2G2T1M7SDW, 
Transfer-Encoding: chunked, 
Connection: close, 
x-amz-id-2: XiUPE6xSNIdSgBWeh/JreYxEjXZlTkMEDoY24l8XyIC1b8eHZiuF0Y/vuSxhOvYUqCslVh8g3II=, 
Date: Mon, 31 Oct 2022 00:10:25 GMT, 
Content-Type: application/xml, 
Server: AmazonS3)), nil)
2022-10-30 17:10:26.741293-0700 Matrix[4233:26088] Swift/ErrorType.swift:200: Fatal error: Error raised at top level: ClientRuntime.SdkError<AWSS3Control.CreateAccessPointOutputError>.client(ClientRuntime.ClientError.retryError(ClientRuntime.SdkError<AWSS3Control.CreateAccessPointOutputError>.service(AWSS3Control.CreateAccessPointOutputError.unknown(AWSClientRuntime.UnknownAWSHttpServiceError(_isThrottling: false, _statusCode: Optional(http_status_400), _headers: Optional(Server: AmazonS3, 
Content-Type: application/xml, 
x-amz-request-id: 5HMBDD2G2T1M7SDW, 
x-amz-id-2: XiUPE6xSNIdSgBWeh/JreYxEjXZlTkMEDoY24l8XyIC1b8eHZiuF0Y/vuSxhOvYUqCslVh8g3II=, 
Transfer-Encoding: chunked, 
Date: Mon, 31 Oct 2022 00:10:25 GMT, 
Connection: close), _message: Optional("Missing required header for this request: x-amz-content-sha256"), _requestID: Optional("5HMBDD2G2T1M7SDW"), _retryable: false, _type: ClientRuntime.ErrorType.unknown)), 
Status Code: http_status_400 
 x-amz-request-id: 5HMBDD2G2T1M7SDW, 
Transfer-Encoding: chunked, 
Connection: close, 
x-amz-id-2: XiUPE6xSNIdSgBWeh/JreYxEjXZlTkMEDoY24l8XyIC1b8eHZiuF0Y/vuSxhOvYUqCslVh8g3II=, 
Date: Mon, 31 Oct 2022 00:10:25 GMT, 
Content-Type: application/xml, 
Server: AmazonS3)), nil)
```

### Expected Behavior

Operation must succeed.

### Current Behavior

Fails with 
```
_message: Optional("Missing required header for this request: x-amz-content-sha256"), _requestID: Optional("5HMBDD2G2T1M7SDW"), _retryable: false, _type: ClientRuntime.ErrorType.unknown)), 
```

### Reproduction Steps

```
 let client = try S3ControlClient(region: "us-west-2")
        let input = CreateAccessPointInput(accountId: "{account id}", bucket: "jangirg-testing", name: "jangirg-testing-ap")
        let output = try await client.createAccessPoint(input: input)
```

### AWS SWIFT SDK version used

0.3.1

### Compiler and Version used

swift-driver version: 1.62.8 Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)

### Operating System and version

arm64-apple-macosx12.0

**Comment by dayaffe** [comment]
I have been able to reproduce this error on the latest SDK version and confirm it still exists: 
`Error: UnknownAWSHTTPServiceError(typeName: Optional("InvalidRequest"), message: Optional("Missing required header for this request: x-amz-content-sha256"), requestID: Optional("EX2K0H9GQ6G5RRD8")`

```
2023-11-07T14:50:56-0500 info HttpContent : [Logging] seeking to offset 0 in data
2023-11-07T14:50:56-0500 info HttpContent : [Logging] read 143 bytes from data
2023-11-07T14:50:56-0500 info HttpContent : [Logging] read 0 bytes from data
2023-11-07T14:50:56-0500 info HttpContent : [Logging] seeking to offset 0 in data
2023-11-07T14:50:56-0500 info S3ControlClient : [Logging] Request: PUT https:443 
 Path: /v20180820/accesspoint/dayaffe-testing-ap 
 Content-Type: application/xml, 
X-Amz-Date: 20231107T195056Z, 
User-Agent: aws-sdk-swift/1.0 ua/2.0 api/s3_control#1.0 os/macos#13.5.2 lang/swift#5.9 cfg/retry-mode#legacy, 
Authorization: AWS4-HMAC-SHA256 Credential=.../20231107/us-west-2/s3/aws4_request, SignedHeaders=content-type;host;x-amz-account-id;x-amz-date, Signature=..., 
Host: 698276607182.s3-control.us-west-2.amazonaws.com, 
x-amz-account-id: ..., 
Content-Length: 143 
 Optional([])
2023-11-07T14:50:56-0500 info SerialExecutor : [Logging] Creating connection pool for 698276607182.s3-control.us-west-2.amazonaws.com with max connections: 50
2023-11-07T14:50:57-0500 info CRTClientEngine : [Logging] Connection was acquired to: Optional("https://....s3-control.us-west-2.amazonaws.com/v20180820/accesspoint/dayaffe-testing-ap?")
2023-11-07T14:50:57-0500 info CRTClientEngine : [Logging] Using HTTP/1.1 connection
2023-11-07T14:50:57-0500 info HttpContent : [Logging] read 143 bytes from data
2023-11-07T14:50:57-0500 info CRTClientEngine : [Logging] Main headers received
2023-11-07T14:50:57-0500 info CRTClientEngine : [Logging] Body chunk received
2023-11-07T14:50:57-0500 info CRTClientEngine : [Logging] Request/response completed
Error: UnknownAWSHTTPServiceError(typeName: Optional("InvalidRequest"), message: Optional("Missing required header for this request: x-amz-content-sha256"), requestID: Optional("EX2K0H9GQ6G5RRD8"), requestID2: nil, httpResponse: 
Status Code: http_status_400 
 Server: AmazonS3, 
Connection: close, 
x-amz-request-id: EX2K0H9GQ6G5RRD8, 
Transfer-Encoding: chunked, 
Content-Type: application/xml, 
Date: Tue, 07 Nov 2023 19:50:57 GMT, 
x-amz-id-2: x6kSXblRRuTdKSwrJvXdTz9SCqOO/qbzIdlqLv29GHq6nZA9OPhUTJHkbiTxuViAyMtnCw+con7O16gH7K23yw==)
Program ended with exit code: 1
```

Will look into this further and update this thread with any findings.

**Comment by jbelkins** [comment]
Released in AWS SDK for Swift 0.32.0
